src/lib/models/document.coffee

```diff
@@ -23,21 +23,78 @@ YAML = null
 # =====================================
 # Classes
 
+###*
+# The DocumentModel class is DocPad's representation
+# of a website or project's content files. This can be
+# individual web pages or blog posts etc. Generally, this
+# is not other website files such as css files, images, or scripts -
+# unless there is a requirement to have DocPad do transformation on
+# these files.
+# Extends the DocPad FileModel class
+# https://github.com/docpad/docpad/blob/master/src/lib/models/file.coffee
+# DocumentModel primarily handles the rendering and parsing of document files.
+# This includes merging the document with layouts and managing the rendering
+# from one file extension to another. The class inherits many of the file
+# specific operations and DocPad specific attributes from the FileModel class.
+# However, it also overrides some parsing and file output operations contained
+# in the FileModel class.
+#
+# Typically we do not need to create DocumentModels ourselves as DocPad handles
+# all of that. Most of the time when we encounter DocumentModels is when
+# querying DocPad's document collections either in the docpad.coffee file or
+# from within a template.
+#
+# 	indexDoc = @getCollection('documents').findOne({relativeOutPath: 'index.html'})
+#
+# A plugin, however, may need to create a DocumentModel depending on its requirements.
+# In such a case it is wise to use the built in DocPad methods to do so, in particular
+# docpad.createModel
+#
+# 	#check to see if the document alread exists ie its an update
+#   docModel = @docpad.getCollection('posts').findOne({slug: 'some-slug'})
+#            
+#   #if so, load the existing document ready for regeneration
+#   if docModel
+#   	docModel.load()
+# 	else
+#      #if document doesn't already exist, create it and add to database
+#      docModel = @docpad.createModel({fullPath:'file/path/to/somewhere'})
+#      docModel.load()
+#      @docpad.getDatabase().add(docModel)
+#
+# @class DocumentModel
+# @constructor
+# @extends require('./file').FileModel
```

Comment:
YUIdoc supports cross linking via:

```
{{#crossLink "Class/method"}}{{/crossLink}}
```

So this could become:

```
{{#crossLink "FileModel"}}{{/crossLink}}
```